grakn-graph/src/main/java/ai/grakn/graph/internal/TypeImpl.java

```diff
@@ -142,6 +142,25 @@ V addInstance(Schema.BaseType instanceBaseType, BiFunction<Vertex, T, V> produce
         return Collections.unmodifiableCollection(filterImplicitStructures(allRoleTypes));
     }
 
+    @Override
+    public Collection<ResourceType> resources() {
+        getGraknGraph().showImplicitConcepts(true); // If we don't set this to true no role types relating to resources will not be retreived
+
+        Set<ResourceType> resourceTypes = new HashSet<>();
+        //A traversal is not used in this caching so that ontology caching can be taken advantage of.
+        playsRoles().forEach(roleType -> roleType.relationTypes().forEach(relationType -> {
+            if(relationType.isImplicit()){
+                //This is faster than doing the traversal
+                TypeName prefix = Schema.Resource.HAS_RESOURCE.getName(TypeName.of(""));
+                TypeName resourceTypeName = TypeName.of(relationType.getName().getValue().replace(prefix.getValue(), ""));
+                resourceTypes.add(getGraknGraph().getType(resourceTypeName));
+            }
+        }));
+
+        getGraknGraph().showImplicitConcepts(false);
```

Comment:
There's a bug here: if the flag was on when entering this method, then it will be off when leaving.
You can add a mark to your "implicit flag" tally!